Change keep alive, deleted jshint errors, fixed reconnectPeriod >= connectTimeout bug

4rzael opened the pull request:
## Change keepalive
Fixes #353 

Let's say that the client has a keepalive of 10 seconds,

Until now, it would send ping every 10 seconds.

Now, if it has received a control packet at 3 seconds, it'll send the next ping at 13s, so it results in a reduced data consumption.

## deleted some jshint errors for node 0.10
Fixes #354 

## reconnectPeriod >= connectTimeout supported
Fixes #346

mcollina commented:
There seems to be a jscs error :).

But great job! Would you like to help contributing to mqtt.js?

4rzael commented:
Okay, I fixed the error.

Also, I changed the version of jshint to 2.7.0 (at least until the node v0.8 compatibility is dropped), so it fixes #354. Is this okay ?

Yeah, I'd love to help contributing to mqtt.js. :)

mcollina commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "mqtt",
   "description": "A library for the MQTT protocol",
-  "version": "1.5.0",
+  "version": "1.5.1",
```
can you please roll this back to 1.5.0. The lead maintainer (myself) usually bump that number.

4rzael replied:
okay !

mcollina commented:
I've given you the commit bit, please follow the CONTRIBUTING.md guide :).
Add yourself to the README as well!

4rzael commented:
Thanks !

I also added another commit to fix #346, should I keep it here, or make another PR ?
